File: 프로그래머스/Lv1/히샤드 수.js
// 내 풀이
function solution(x) {
    let num = 0;
    x = x.toString();
    
    for(let i=0; i<x.length;i++){
        num += Number(x[i]);
    }
    return x % num === 0 ? true : false;
}

// 다른 사람 풀이
function solution2(x){
    return !(x % (x+"").split("").reduce((a, c) => +a + +c));
}

console.log(solution(10));
console.log(solution(12));
console.log(solution2(11));
console.log(solution2(13));

File: 프로그래머스/Lv1/히샤드 수.py
# 내 코드
def solution(x):
    answer = True
    num = 0

    for i in str(x):
        num += int(i)

    if x % num == 0: 
        answer = True 
    else: 
        answer = False

    return answer

# 다른 사람 코드
def solution2(x):
    return x%sum([int(i) for i in str(x)]) == 0

print(solution(10))
print(solution(12))
print(solution2(11))
print(solution2(13))